addOns/kotlin/src/test/java/org/zaproxy/zap/extension/kotlin/VerifyScriptTemplates.java

```diff
@@ -0,0 +1,110 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.kotlin;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.Reader;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import javax.script.Compilable;
+import javax.script.CompiledScript;
+import javax.script.ScriptContext;
+import javax.script.ScriptEngine;
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.zaproxy.zap.extension.alert.ExtensionAlert;
+import org.zaproxy.zap.testutils.AbstractVerifyScriptTemplates;
+
+/** Verifies that the Kotlin script templates are parsed without errors. */
+public class VerifyScriptTemplates extends AbstractVerifyScriptTemplates {
+
+    private static ScriptEngine se;
+
+    @BeforeAll
+    public static void setUp() {
+        se = new KotlinEngineWrapper(Thread.currentThread().getContextClassLoader()).getEngine();
+    }
+
+    @Override
+    protected String getScriptExtension() {
+        return ".kts";
+    }
+
+    @Override
+    protected void parseTemplate(Path template) throws Exception {
+        try (Reader reader = Files.newBufferedReader(template, StandardCharsets.UTF_8)) {
+            Compilable c = (Compilable) se;
+            CompiledScript cs = c.compile(reader);
+            cs.eval();
+        }
+    }
+
+    @Test
+    public void shouldPrintToScriptContextWriter() throws Exception {
+        String script = getScriptContents("printTest.kts");
+
+        ByteArrayOutputStream console = new ByteArrayOutputStream();
+        ScriptContext sc = se.getContext();
+        Writer originalWriter = sc.getWriter();
+
+        try {
+            sc.setWriter(new PrintWriter(console));
+            Compilable c = (Compilable) se;
+            CompiledScript cs = c.compile(script);
+
+            cs.eval();
+
+            sc.getWriter().flush();
+            assertEquals("ZAP" + System.lineSeparator(), console.toString());
+        } finally {
+            sc.setWriter(originalWriter);
+        }
+    }
+
+    @Test
+    public void shouldHaveSameHashCode() throws Exception {
+
+        int jvmHashCode = ExtensionAlert.class.hashCode();
```

Comment:
- It needs to use other class than the ones available in the context class loader (technically it can be the same class file but loaded from other class loader), otherwise we don't know if it was from the provided class loader or the context class loader (the class loaders need to be different).
 - Better return (or pass through the bindings) the `Class` and check by reference, to not rely on the hash code which can be changed any time.

Reply:
Ok I think I understand what you're saying and I suspected there was a better way to do this. Let me see if I can create a better test setup for this.